tools/rewriter/GenCallAsm.cpp

```diff
@@ -280,6 +286,335 @@ static std::string sig_string(const CAbiSignature &sig,
   return ss.str();
 }
 
+static void emit_fn_call(
+    const std::optional<std::string> target_name,
+    WrapperKind kind,
+    AsmWriter &aw,
+    Arch arch) {
+
+  add_comment_line(aw, "Call wrapped function");
+
+  if (!target_name) {
+    // indirect call
+    assert(kind == WrapperKind::IndirectCallsite);
+    if (arch == Arch::X86) {
+      add_asm_line(aw, "call *%r12");
+    } else if (arch == Arch::Aarch64) {
+      add_asm_line(aw, "br x12"); // TODO, which register?
+    }
+  } else {
+    // direct call
+    if (arch == Arch::X86) {
+      add_asm_line(aw, "call "s + target_name.value());
+    } else if (arch == Arch::Aarch64) {
+      add_asm_line(aw, "b "s + target_name.value());
+    }
+  }
+}
+
+static AsmWriter get_asmwriter(bool as_macro) {
+  std::string terminator = {};
+  // Code generated as a macro needs to terminate each line with '\'
+  if (as_macro) {
+    terminator = "\\"s;
+  }
+  return {.ss = {}, .terminator = terminator};
+}
+
+static void emit_save_regs(AsmWriter &aw, uint32_t caller_pkey, uint32_t target_pkey, Arch arch) {
```

Comment:
I think `emit_prologue` and `emit_epilogue` make more sense than `emit_save_regs` and `emit_restore_regs` since there are other places where we save and restore regs (e.g. across calls to libia2_scrub_registers).

Reply:
Done.